argocd/resource_argocd_repository.go

```diff
@@ -40,62 +42,92 @@ func resourceArgoCDRepositoryCreate(d *schema.ResourceData, meta interface{}) er
 	tokenMutexConfiguration.Unlock()
 
 	if err != nil {
-		return err
+		return []diag.Diagnostic{
+			{
+				Severity: diag.Error,
+				Summary:  fmt.Sprintf("Repository %s not found", repo),
+				Detail:   err.Error(),
+			},
+		}
 	}
 	if r == nil {
-		return fmt.Errorf("ArgoCD did not return an error or a repository result")
+		return []diag.Diagnostic{
+			{
+				Severity: diag.Error,
+				Summary:  fmt.Sprintf("ArgoCD did not return an error or a repository result"),
+			},
+		}
 	}
 	if r.ConnectionState.Status == application.ConnectionStatusFailed {
-		return fmt.Errorf(
-			"could not connect to repository %s: %s",
-			repo.Repo,
-			r.ConnectionState.Message,
-		)
+		return []diag.Diagnostic{
+			{
+				Severity: diag.Error,
+				Summary: fmt.Sprintf(
+					"could not connect to repository %s: %s",
+					repo.Repo,
+					r.ConnectionState.Message,
+				),
+			},
+		}
 	}
 	d.SetId(r.Repo)
-	return resourceArgoCDRepositoryRead(d, meta)
+	return resourceArgoCDRepositoryRead(ctx, d, meta)
 }
 
-func resourceArgoCDRepositoryRead(d *schema.ResourceData, meta interface{}) error {
+func resourceArgoCDRepositoryRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	server := meta.(ServerInterface)
 	c := *server.RepositoryClient
 	r := &application.Repository{}
 
 	featureRepositoryGetSupported, err := server.isFeatureSupported(featureRepositoryGet)
 	if err != nil {
-		return err
+		return []diag.Diagnostic{
+			{
+				Severity: diag.Error,
+				Summary:  `support for feature "repositoryGet" could not be checked`,
+				Detail:   err.Error(),
+			},
+		}
 	}
 
-	switch featureRepositoryGetSupported {
-	case true:
+	if featureRepositoryGetSupported {
 		tokenMutexConfiguration.RLock()
-		r, err = c.Get(context.Background(), &repository.RepoQuery{
+		r, err = c.Get(ctx, &repository.RepoQuery{
 			Repo:         d.Id(),
 			ForceRefresh: true,
 		})
 		tokenMutexConfiguration.RUnlock()
 
 		if err != nil {
-			switch strings.Contains(err.Error(), "NotFound") {
-			// Repository has already been deleted in an out-of-band fashion
-			case true:
+			if strings.Contains(err.Error(), "NotFound") {
```

Comment:
Same thing for style consistency, keep the switch block within if block

Reply:
Using `switch` with booleans is quite unusual.

Reply:
Right..

Reply:
👍